docs/source/tutorial-pyfunc.rst

```diff
@@ -0,0 +1,368 @@
+
+Add the following code to **train.py**. Note that you must *add* the line ``run_id = store_pyfunc_model(classifier, signature)`` to the entry point.
+
+.. code-block:: python
+
+    class IrisClassifier(mlflow.pyfunc.PythonModel):
+        def load_context(self, context : mlflow.pyfunc.PythonModelContext) -> None:
+            self.classifier = jl.load(context.artifacts["classifier"])
+
+        def predict(self, context : mlflow.pyfunc.PythonModelContext, model_input : pd.DataFrame) -> np.ndarray[int]:
+            preds = self.classifier.predict(model_input)
+            return iris_classes(preds)
+
+
+    def store_pyfunc_model(classifier : LogisticRegression, signature : ModelSignature) -> str:
+        # MLflow Tracking URI *must* be http or https for log_model
+        assert(mlflow.get_tracking_uri().startswith("http"))
+
+        classifier_filename = "iris_classifier.joblib"
+        jl.dump(classifier, classifier_filename)
+        artifacts = {
+            "classifier": classifier_filename
+        }
+
+        model_info = mlflow.pyfunc.log_model(
+            python_model=IrisClassifier(),
+            artifacts=artifacts,
+            artifact_path="artifacts",
+            signature=signature,
+            code_path= ["custom_code.py"]
+        )
+        print(f"Stored in mlruns/{model_info.run_id}")
+        return model_info.run_id
+
+    if __name__ == "__main__":
+        with mlflow.start_run() as run:
+            classifier, signature = initialize_and_train()
+            run_id = store_pyfunc_model(classifier, signature)
+
+
+Starting at the entry point, you can see that the code does not change the creation of ``classifier`` and ``signature``. These values are passed to ``store_pyfunc_model``. 
+
+The first thing that ``store_pyfunc_model`` does is confirm that the MLflow Tracking Server is running on a URI that starts with ``http`` or ``https``, as discussed in :ref:`setup`.
+
+The code then uses the ``joblib`` library to serialize the model to disk. Using serialization/deserialization to save and load the model and related files can make for clear and efficient code, but it is not required. You can store any type of file in the ``artifacts`` directory: JSON files, binary weight files, vector database files, and so forth. 
+
+The ``artifacts`` dictionary is then created. Its ``classifier`` key is set to the name of the file containing the serialized model: ``"iris_classifier.joblib"``.
+
+An instance of the ``IrisClassifier`` class is then instantiated. This class derives from :py:class:`mlflow.pyfunc.PythonModel`. As discussed at the beginning of this section, this class implements the ``load_context`` and ``predict`` methods. The ``IrisClassifier`` implementation of those methods will be discussed in the next section. 
```

Comment:
`mlflow.pyfunc.PythonModel` is an ABC. It defines the abstract methods `load_context` and `predict`, but doesn't implement them.